src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -2437,22 +2464,100 @@ private static MethodInfo GetCurryClosureMethod(object lambda, bool isAction)
                 return invokeMethod != null && EmitMethodCall(il, invokeMethod);
             }
 
+            private static bool TryEmitInvertedNullComparison(Expression expr,
+                IReadOnlyList<ParameterExpression> paramExprs, ILGenerator il, ref ClosureInfo closure)
+            {
+                if (expr is BinaryExpression b)
+                {

```

Comment:
Could we collapse this thingy a bit, many `return false` inside can be combined to reduce noize.

Reply:
Do you have a good idea?